src/sql/src/plan/statement/ddl.rs

```diff
@@ -735,6 +764,86 @@ pub fn plan_create_source(
             (connection, encoding, None)
         }
     };
+
+    let (available_subsources, requested_subsources) = match (available_subsources, subsources) {
+        (Some(available_subsources), Some(CreateReferencedSubsources::Subset(subsources))) => {
+            let mut requested_subsources = vec![];
+            for subsource in subsources {
+                let name = subsource.reference.clone();
+
+                let target = match &subsource.subsource {
+                    Some(subsource) => match subsource {
+                        DeferredObjectName::Named(target) => target.clone(),
+                        DeferredObjectName::Deferred(name) => {
+                            // TODO: remove this after the next release, we need
+                            // it only so we can load the catalog to do the
```

Comment:
Since you're here, can we remove this now? If it's more complicated than just deleting these lines feel free to ignore this comment and doing a separate PR

Reply:
Sadly the upgrade framework doesn't reflect our current upgrade process (i.e. it takes every version to the current version and does not presume a linear path along versions), which makes this fail. Will file a ticket about this.